### DISCUSSION TITLE
Customizable 24 words

### DISCUSSION THREAD

**Discussion by Frodooio** [iotaledger/firefly#1585] [Features / Enhancements]
## Description

Just like was possible on creation of a seed in the Trinity wallet, allow changing the 24 word mnemonic by clicking on words to switch to another randomly chosen word from the dictionary list.

## Motivation

Extra security for true randomness.

## Are you planning to do it yourself in a pull request?

No.

**Comment by UmbahaSpodPaha** [comment]
Actually there is "true randomness" from 2048 words (bip39 list). You won't get any better than that yourself. You can't actually do a valid 24words mnemonic with the words you choose yourself.

**Comment by ghost** [comment]
https://youtu.be/hRXcY_tIlrw?t=669
no need to add random new words  imo

**Comment by Frodooio** [comment]
I'm talking about the randomness that you now derive only from the computer (to choose the words from the list). If there is ever a fault detected in the randomness method on OSX or windows, that could put the seed at risk, couldn't it? So we add to randomness by letting the user choose if he wants to change some of the words (or not!) This was done in the same way in Trinity, for probably the same reasons?

**Comment by ghost** [comment]
![image](https://user-images.githubusercontent.com/48189056/116704905-3b679a00-a9cc-11eb-9230-40ed7f012347.png)
i think they use your password as an extra passphrase  added to your 24 adding randomizing so even if there is a problem with the randomizing of the 24 words there is still password added protection  if they can find a better way tot guess the 24 words due to bad randomizing. . but i'm just guessing here since i didnt code it.

**Comment by Frodooio** [comment]
Oh if they did that would be great, would love to hear a confirmation from devs. thx!

**Comment by Frodooio** [comment]
Actually now that I think about it, no the password isn't added because the only thing needed to recover your account is your 24 words. You don't need to remember the password at all..

**Comment by nothingismagick** [comment]
Hi, I'll try to explain how this works without going too deep into the weeds of cryptography or security. 

We use the best possible source of random available to the system on your device, so don't worry about randomness. Adding new words to the BIP39 wordlist would actually be wrong from the perspective of the specification, and would break the system. 

This mnemonic is like the DNA of your account. It is what is used to derive keys for all your transactions and is absolutely essential for you to keep secret and NEVER SHARE WITH ANYONE.

The password is used to encrypt / decrypt the Stronghold snapshot, and does not modify your mnemonic phrase. It keeps the mnemonic safe and secure within Stronghold.

**Comment by Frodooio** [comment]
Its not about adding new words, just allowing the user to click on the word to change it to another random word. (Same as trinity allowed to click on a letter of the seed and it changed). Not sure why such a feature couldn't be added. Oh well.

**Comment by nothingismagick** [comment]
Ah - then I misunderstood. SRY
